Parser/tag/mappers

caroo7 opened the pull request:
Tag mappers. Retrieving tags when they occured.

caroo7 committed:
Tag integration. Work in progress.

caroo7 committed:
Tag integration. Work in progress.

Commit:
Retrieving tags finished.
If there is no tags (like in empik) just return empty set.

Change-Id: I39fcdb83b9ed5b4a25d3244b4fb887b9f44f0984

Commit:
Retrieving tags finished.
If there is no tags (like in empik) just return empty set.

Change-Id: I39fcdb83b9ed5b4a25d3244b4fb887b9f44f0984

sledzgrzegorz commented:
I have looked at this code and I think it is ok.

LukaszCzubernat commented on parser/src/main/java/converter/BookDetailsToBookAssembler.java:
```diff
@@ -116,15 +117,14 @@ private Book convert(BookDetails bookDetails) {
         }
-        return tags;
+        return bookDetails.getTags().stream().map(s -> tagRepo.findByName(tagMapper.getMap().get(s)))
+                .filter(tag -> tag != null)
```
Maybe use Objects.nonNull()

LukaszCzubernat commented on parser/src/main/java/converter/BookDetailsToBookAssembler.java:
```diff
@@ -116,15 +117,14 @@ private Book convert(BookDetails bookDetails) {
-        Tag tag = tagRepo.findByName(tagString);
-        if(tag != null) {
-            tags.add(tag);
+        if (bookDetails.getTags() == null) {
```
Optional?
Or better add Optional to your last 'return' and there filter books without tags

LukaszCzubernat commented on parser/src/main/java/converter/BookDetailsToBookAssembler.java:
```diff
@@ -84,8 +85,8 @@ private Book convert(BookDetails bookDetails) {
-        }else {
-            log.debug("Book will be saved on database: "+book);
+        } else {
+            log.debug("Book will be saved on database: " + book);
```
if-else just for logging?

LazyScoundrel committed:
Changes in the GUI

I have made various changes so that now we can switch between the views of Publio and Empik. For now the values are hardcoded, and if another library will be added I will need to make$
more generic based on the currently showed table.
This works now for presentation purposes and will be changed in final product.
This branch can be merged for now.

caroo7 committed:
Merge pull request #38 from caroo7/COMPONENT/GUI/PUBLIO

Changes in the GUI

caroo7 committed:
Working on assembler changing.
Still in progress.

Commit:
Merge branch 'parser/tag/mappers' of https://github.com/caroo7/Caro-robot into parser/tag/mappers

Conflicts:
	cache/src/main/java/access/DBAccessor.java
	cache/src/main/java/cache/Cache.java
	parser/src/main/java/converter/BookDetailsToBookAssembler.java
	parser/src/main/java/main/ParserMain.java

Change-Id: I253a0f2ec0789ff14adaf6655e30d65f0ad0e30f

Commit:
Finished implementation of one parser.
Right now parser generates one cache file for each library.
Improvements in tag retrieving after code review.

Change-Id: I56fac63ee2cc5f09532492307888fad64462ef88

Commit:
README improved.

Change-Id: I05c6dbe30b08016c71df5aac0abc01134b0214be

LukaszCzubernat commented:
ok :+1:

LazyScoundrel commented on cache/src/main/java/access/CacheReader.java:
```diff
@@ -12,10 +12,10 @@
 
         List<Book> books = new ArrayList<>();
-        try (FileInputStream fileInputStream = new FileInputStream("cache.txt")) {
+        try (FileInputStream fileInputStream = new FileInputStream(libName)) {
```
:+1:

LazyScoundrel commented on database/src/main/java/entities/Library.java:
```diff
@@ -16,23 +17,16 @@
+    @Getter
     private String name;
-
+    @Getter
```
:+1:

LazyScoundrel commented on database/src/main/java/entities/Tag.java:
```diff
@@ -11,11 +10,12 @@
 import java.io.Serializable;
 
 @Entity
+@ToString
```
Do we really want the Lombok's ToString here? You know it's format and it won't be good in terms of viewing

LazyScoundrel commented on database/src/main/java/repositories/BookRepository.java:
```diff
@@ -11,5 +12,6 @@
     List<Book> findAll();
 
-}
+    List<Book>  findByLibrary(Library library);
```
:+1:

LazyScoundrel commented on parser/src/main/java/parser/empik/EmpikBookParser.java:
```diff
@@ -23,7 +24,7 @@ public String getTitle(Optional<Document> document) {
             return  document.get().select("span.pDAuthorList > a").stream().map(element -> element.text()).collect(Collectors.toSet());
         }
-        return null;
+        return Collections.emptySet();
```
:+1:

LazyScoundrel commented:
Overall these are good changes :)

caroo7 committed:
Improvements in toString method. Don't use lombok for this right now.
Some restriction only for demo purposes.
